lib/otp_fetcher.ex

```diff
@@ -49,10 +51,19 @@ defmodule Burrito.OTPFetcher do
 
     {_, download_url} = selected_version
 
-    Logger.info("Downloading replacement ERTS: #{download_url}")
+    cache_key = :crypto.hash(:sha, to_string(erts_version) <> to_string(otp_version) <> to_string(platform)) |> Base.encode16()
+
+    case FileCache.fetch(cache_key) do
+      {:hit, data} ->
+        Logger.info("Found matching cached ERTS, using that")
+        do_unpack(data, release_path, erts_version, platform)
+      {:miss, _} ->
+        Logger.info("Downloading replacement ERTS: #{download_url}")
+        data = Req.get!(download_url).body
```

Comment:
using Req built-in, also fs-based, cache should be as simple as just:

```suggestion
        data = Req.get!(download_url, cache: true).body
```

it would put it under $user_cache/req though, currently there's no way to customise it but we could definitely change that.

Reply:
Oh that's awesome! I might just leverage that if you think we can add the option to pass in a custom directory later!

Reply:
yeah, definitely. I'm thinking instead of the usual `cache: true`, one could pass: `cache: [path: ...]`.

I forgot to mention that req cache isn't totally equivalent to your solution, req would set the if-modified-since header and _hit_ the server (hoping for http 304). That's probably ok but definitely less efficient than not hitting the server at all.

Reply:
Running some tests using the Req cache it definitely still benefits from the speedup! Given that caveat I might still continue with this simple module implementation for now, as I want to also provide totally offline builds if possible.

but the req cache is certainly a future consideration!